mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/AnnotationHandlerMapping.java

```diff
@@ -14,16 +23,69 @@ public class AnnotationHandlerMapping {
     private final Object[] basePackage;
     private final Map<HandlerKey, HandlerExecution> handlerExecutions;
 
-    public AnnotationHandlerMapping(final Object... basePackage) {
+    public AnnotationHandlerMapping(final Object... basePackage) throws Exception {
         this.basePackage = basePackage;
         this.handlerExecutions = new HashMap<>();
+        initialize();
     }
 
-    public void initialize() {
+    public void initialize() throws Exception {
+        for (Class<?> controllerClass : findControllerClasses()) {
+            initializeHandlerExecutions(controllerClass);
+        }
+
         log.info("Initialized AnnotationHandlerMapping!");
     }
 
     public Object getHandler(final HttpServletRequest request) {
-        return null;
+        String url = request.getRequestURI();
+        RequestMethod requestMethod = RequestMethod.from(request.getMethod());
+
+        HandlerKey handlerKey = new HandlerKey(url, requestMethod);
+        return handlerExecutions.get(handlerKey);
+    }
```

Comment:
```suggestion
        RequestMethod requestMethod = RequestMethod.from(request.getMethod());
        String url = request.getRequestURI();
        
        HandlerKey handlerKey = new HandlerKey(url, requestMethod);
        return handlerExecutions.get(handlerKey);
    }
```

위 순서 중 from 은 예외를 발생시킬 수 있는 함수네요. 메서드에서 필요한 정보를 초기화 하는 작업 중 예외가 발생할 수 있는 상황들은 가능한 앞쪽에 배치하는게 좋을 수 있겠습니다. 위 메서드에서는 사소한 부분이긴 하지만 항상 예외는 빠르게 발생시킬 수록 불필요한 작업을 줄일 수 있으니까요!

```java
String url = request.getRequestURI();
```
그리고 마찬가지로 사소한 부분입니다만 리퀘스트에서 uri를 가져오는 변수의 이름은 url이네요.
아래의 `addMappings` 메서드에서는 `HandlerKey`의 인자로 넘어가는 변수의 이름이 uri 로 적혀있구요. 
uri와 url은 비슷하지만 확실히 다릅니다. 통일성의 문제도 있겠지만 요청 매핑 작업에서 필요한 것이 uri인지 url인지는 생각해볼 지점일 것 같아요!

Reply:
1. 저도 예외를 빠르게 발생시키는 방식을 좋아해요! 사소한 부분인데 꼼꼼히 봐주셔서 감사합니다! 😄 
2. URI는 식별자, URL은 자원의 위치를 나타내는군요. RFC2616에서도 Request **URI**라고 표시하고 있기 때문에 URI가 더 적절하겠네요.